hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/table/format/cow/CopyOnWriteInputFormat.java

```diff
@@ -99,10 +113,36 @@ public CopyOnWriteInputFormat(
     this.selectedFields = selectedFields;
     this.conf = new SerializableConfiguration(conf);
     this.utcTimestamp = utcTimestamp;
+    this.schemaEvolutionContext = SchemaEvolutionContext.of(flinkConf);
   }
 
   @Override
   public void open(FileInputSplit fileSplit) throws IOException {
+    String[] actualFieldNames;
+    DataType[] actualFieldTypes;
+    if (schemaEvolutionContext.isPresent()) {
+      SchemaEvolutionContext context = schemaEvolutionContext.get();
+      InternalSchema actualSchema = context.getActualSchema(fileSplit);
```

Comment:
Just make the `fullFieldNames` and `fullFieldTypes` as ones after schema evolution, and move these schema evolution components/logic to separate tool clazz.

Reply:
The same https://github.com/apache/hudi/pull/5830#discussion_r927284257

Reply:
Moved this logic to separate method `setActualFields`